server/service/service_test.go

```diff
@@ -244,16 +247,16 @@ func createTestClient(t *testing.T, addr string, tlsClientCreds credentials.Tran
 	return conn
 }
 
-// Dummy TLS config for tests
-func testClientTLSConfig(t *testing.T) credentials.TransportCredentials {
-	t.Helper()
-	// In a real test, you'd use a cert signed by a CA the server trusts.
-	// For simplicity, using insecure here, as the server logic *rejects* client certs.
-	return insecure.NewCredentials()
-}
-
 func TestBootstrapStream(t *testing.T) {
-	goodCert := base64.StdEncoding.EncodeToString(createTestCertificate(t, "test-device", "test-serial-123"))
+	devicePrivKey, goodCertDER := createTestCertificate(t, "test-device", "test-serial-123")
+	goodCert := base64.StdEncoding.EncodeToString(goodCertDER)
+	// Create a bootloader config struct for the test case.
+	bootloaderConfig, err := structpb.NewStruct(map[string]interface{}{
```

Comment:
The bootloader config is actually not a commonly implemented field for vendors. For testing here, I'd recommend instead using "vendor_config". This is saves having to make that Struct type.

Reply:
good catch.